public/app/features/alerting/unified/hooks/useIsRuleEditable.ts

```diff
@@ -16,19 +16,22 @@ interface ResultBag {
 }
 
 export function useIsRuleEditable(rulesSourceName: string, rule?: RulerRuleDTO): ResultBag {
-  const { currentData: dsFeatures, isLoading } = featureDiscoveryApi.endpoints.discoverDsFeatures.useQuery({
-    uid: getDatasourceAPIUid(rulesSourceName),
-  });
+  const { currentData: dsFeatures, isLoading: loadingDataSourceFeatures } =
+    featureDiscoveryApi.endpoints.discoverDsFeatures.useQuery({
+      uid: getDatasourceAPIUid(rulesSourceName),
+    });
 
   const folderUID = rule && isGrafanaRulerRule(rule) ? rule.grafana_alert.namespace_uid : undefined;
-
   const rulePermission = getRulesPermissions(rulesSourceName);
-  const { folder, loading } = useFolder(folderUID);
+
+  const { folder, loading: loadingFolder } = useFolder(folderUID);
```

Comment:
We're still using redux here, which should be refactored ASAP as it will cache a `folderDTO` until the entire browser session is refreshed.

Reply:
Have a PR up for that 👉 https://github.com/grafana/grafana/pull/100645